### PR TITLE
Add variadic node guide articles

### LINKED ISSUE #1069: Describe variadic nodes and arity concepts in docs
There should be guide articles on:

- Using variadic nodes: what is arity, how they are expanded
- Creating a variadic node: marker nodes, requirements for terminals

### PR DESCRIPTION
**Pull request by nkrkv**
Closes #1069 

Also:

- Move the Concepts section higher
- Remove outdated tutorials from index

### CODE REVIEW

File: docs/guide/creating-variadics/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+![Join node implementation](./join.patch.png)
+
+## Rules
+
+To let XOD properly expand your variadic node when transpiling you must satisfy
+several natural restrictions:
+
+1. There should be no more than a single `variadic-*` marker.
+2. A variadic patch node should have at least one output.
+3. Let a patch node has 𝜄 inputs, 𝜔 outputs, and arity step 𝜎.
```
Comment (brusherru) [inline docs/guide/creating-variadics/README.md]: IMHO, It's too complicated.
Maybe just: "Amount of inputs should be greater or equal to a sum of outputs and arity step" or something like this?

File: docs/guide/creating-variadics/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   The following must be true: 𝜄 ≥ 𝜔 + 𝜎
+4. The types of inputs from the middle (indexes 𝜄 − 𝜎 − 𝜔 through 𝜄 − 𝜎 − 1)
```
Comment (brusherru) [inline docs/guide/creating-variadics/README.md]: 🤕 
If User already knows about "shared pin" and "variadic pin" terms, we could introduce term "middle pins" (or accumulation pins), that in the middle of shared and variadic. And input and output pairs of the "middle pins" must be of the same type even if an implicit cast is allowed in other cases.

Or something like this. :)

**Comment by nkrkv** [comment]
@brusherru Removed the awkward math. Take a look.

File: docs/guide/creating-variadics/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+**Accumulator pins**. Pins in the middle get this role. Their number equals the
+output number. In the example they are `A1`, `A2`, and `A3`. The user provides
+values only for the top level of the cascade; all other levels get accumulator
+values straightly from output pins of the respective upstream node.
+
+**Shared pins**. The leftmost pins which are not variadic nor accumulator get
+this role. In the example they are `S1` and `S2`. The values are provided by
+the user, and they are delivered to each level in the cascade as is. In many
+cases there would be null shared pins.
```
Comment (evgenykochetkov) [inline docs/guide/creating-variadics/README.md]: "null"?

Comment (nkrkv) [reply docs/guide/creating-variadics/README.md]: :thinking: zero? nil?

Comment (evgenykochetkov) [reply docs/guide/creating-variadics/README.md]: IMHO, "zero" is fine. Or even "there would be **no** shared pins"

File: docs/guide/variadics/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+more or fewer values.
+
+In XOD IDE the variadic nodes have a little gripper on the right, and the pins
+which can scale are marked with the ellipsis (⋯) on mouse hover. Depending on
+requirements you can add groups of variadic pins up to an infinite number. Drag
+the right border of the node to adjust.
+
+![Overview](./overview.patch.png)
+
+The variadic pins replication count is called *arity level*. By default, the
```
Comment (evgenykochetkov) [inline docs/guide/variadics/README.md]: "The variadic pins replication count" looks a bit scary, but I can't come up with a better alternative. @brusherru, what do you think?

Comment (nkrkv) [reply docs/guide/variadics/README.md]: The number of variadic pin copies is called arity level?

File: docs/guide/variadics/README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+implementations are induced automatically by XOD. To achieve it, when
+transpiling the program each particular variadic node passes an *expansion*
+stage in which the system replaces it with a cascade of primitive level-1
+nodes. The replacement principle is better illustrated with an example.
+
+![Expansion](./expansion.patch.png)
+
+<div class="ui segment note">
+<span class="ui ribbon label">Pro</span>
+Experienced developers with functional programming experience could note the
```
Comment (evgenykochetkov) [inline docs/guide/variadics/README.md]: "**Experienced** developers with functional programming **experience**"

File: docs/guide/variadics/README.md (hunk 4 above)
Comment (brusherru) [reply docs/guide/variadics/README.md]: > The number of variadic pin copies is called arity level?

I like the original variant more :)

File: docs/guide/variadics/README.md (hunk 5 above)
Comment (nkrkv) [reply docs/guide/variadics/README.md]: Savvy developers with functional programming experience?

Comment (evgenykochetkov) [reply docs/guide/variadics/README.md]: I vote for "Developers with functional programming experience" 🙂

**Comment by nkrkv** [comment]
All’s done

**Pull request merged**